packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -173,24 +177,35 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
         <div
           style={maxRows != null && tagState.showCollapseButton && isCollapsed ? {maxHeight: tagState.maxHeight, overflow: 'hidden'} : undefined}
           ref={containerRef}
-          className={classNames(styles, 'spectrum-Tags-container')}>
-          <div
-            ref={tagsRef}
-            {...gridProps}
-            className={classNames(styles, 'spectrum-Tags')}>
-            {visibleTags.map(item => (
-              <Tag
-                {...item.props}
-                key={item.key}
-                item={item}
-                state={state}
-                allowsRemoving={allowsRemoving}
-                onRemove={onRemove}>
-                {item.rendered}
-              </Tag>
-            ))}
-          </div>
-          {showActions &&
+          className={
+            classNames(
+              styles,
+              'spectrum-Tags-container',
+              {
+                'spectrum-Tags-container--empty': isEmpty
+              }
+            )
+          }>
+          <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
+            <div
+              ref={tagsRef}
+              {...gridProps}
+              className={classNames(styles, 'spectrum-Tags')}>
+              {visibleTags.map(item => (
+                <Tag
+                  {...item.props}
+                  key={item.key}
+                  item={item}
+                  state={state}
+                  allowsRemoving={allowsRemoving}
+                  onRemove={onRemove}>
+                  {item.rendered}
+                </Tag>
+              ))}
+              {isEmpty && renderEmptyState()}
```

Comment:
Noticed that focus is being lost to the body when all the tags are removed, is that going be handled in this PR or a separate one?

Reply:
It should now go to the container when the last tag is removed.